pkgs/applications/misc/slic3r/prusa3d.nix

```diff
@@ -72,8 +79,18 @@ stdenv.mkDerivation rec {
     XMLSAX
   ]);
 
+  # The build system uses custom logic for finding the nlopt library, which
+  # doesn't work for paths in the nix store. We need to set it manually.
+  NLOPT = "${nlopt}";
+
   prePatch = ''
+    # In nix ioctls.h isn't available from the standard kernel-headers package
+    # on other distributions. As the copy in glibc seems to be identical to the
+    # one in the kernel, we use that one instead.
     sed -i 's|"/usr/include/asm-generic/ioctls.h"|<asm-generic/ioctls.h>|g' xs/src/libslic3r/GCodeSender.cpp
+
+    # PERL_VENDORARCH and PERL_VENDORLIB aren't detected correctly by the build
+    # system, so we have to override them
```

Comment:
Why are these overrides done differently from the NLOPT override?

Reply:
Those variables aren't picked up by the build system when set as environment variables, while there is explict support for setting NLOPT in `xs/src/libnest2d/cmake_modules/FindNLopt.cmake`

Reply:
I've amended the comments to hopefully clarify the issue